fe/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java

```diff
@@ -153,122 +178,129 @@ public int getDesiredConcurrentNum() {
         return desiredConcurrentNum;
     }
 
-    public int getMaxErrorNum() {
+    public long getMaxErrorNum() {
         return maxErrorNum;
     }
 
-    public String getKafkaEndpoint() {
-        return kafkaEndpoint;
+    public long getMaxBatchIntervalS() {
+        return maxBatchIntervalS;
+    }
+
+    public long getMaxBatchRows() {
+        return maxBatchRows;
+    }
+
+    public long getMaxBatchSize() {
+        return maxBatchSizeBytes;
+    }
+
+    public String getKafkaBrokerList() {
+        return kafkaBrokerList;
     }
 
     public String getKafkaTopic() {
         return kafkaTopic;
     }
 
-    public List<Integer> getKafkaPartitions() {
-        return kafkaPartitions;
+    public List<Pair<Integer, Long>> getKafkaPartitionOffsets() {
+        return kafkaPartitionOffsets;
     }
 
     @Override
-    public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
+    public void analyze(Analyzer analyzer) throws UserException {
         super.analyze(analyzer);
+        // check dbName and tableName
+        checkDBTable(analyzer);
         // check name
         FeNameFormat.checkCommonName(NAME_TYPE, name);
-        // check dbName and tableName
-        checkDBTableName();
         // check load properties include column separator etc.
         checkLoadProperties(analyzer);
-        // check routine load properties include desired concurrent number etc.
-        checkRoutineLoadProperties();
-        // check custom properties
-        checkCustomProperties();
+        // check routine load job properties include desired concurrent number etc.
+        checkJobProperties();
+        // check data source properties
+        checkDataSourceProperties();
     }
 
-    private void checkDBTableName() throws AnalysisException {
-        if (Strings.isNullOrEmpty(dbTableName.getDb())) {
-            String dbName = ConnectContext.get().getDatabase();
-            if (Strings.isNullOrEmpty(dbName)) {
-                throw new AnalysisException("please choose a database first");
-            }
-            dbTableName.setDb(dbName);
-        }
-        if (Strings.isNullOrEmpty(dbTableName.getTbl())) {
-            throw new AnalysisException("empty table name in create routine load statement");
+    public void checkDBTable(Analyzer analyzer) throws AnalysisException {
+        labelName.analyze(analyzer);
+        dbName = labelName.getDbName();
+        name = labelName.getLabelName();
+        if (Strings.isNullOrEmpty(tableName)) {
+            throw new AnalysisException("Table name should not be null");
         }
```

Comment:
db and table is't checked if it exists?

Reply:
It will be check in `RoutineLoadManager.fromCreateStmt()`